CONTRIBUTING.md

```diff
@@ -1,8 +1,78 @@
 # JFrog welcomes community contribution!
 
-Before we can accept your contribution, process your GitHub pull requests, and thank you full heartedly, we request that you will fill out and submit JFrog's Contributor License Agreement (CLA).
+Before we can accept your contribution, process your GitHub pull requests, and thank you full-heartedly, we request that you will fill out and submit JFrog's Contributor License Agreement (CLA).
 
 [Click here](https://secure.echosign.com/public/hostedForm?formid=5IYKLZ2RXB543N) to submit the JFrog CLA.
 This should only take a minute to complete and is a one-time process.
 
 *Thanks for Your Contribution to the Community!* :-)
+
+## Pull Request Process ##
+
+- Fork this repository.
+- Clone the forked repository to your local machine and perform the proposed changes. You can use [artifactory-user-plugins-devenv](https://github.com/JFrogDev/artifactory-user-plugins-devenv) to setup your development environment.
+- Make sure your changes follow the Acceptance Criteria below.
+- When you are confident about the changes, create a Pull Request pointing to the master branch of this repository. 
+
+## Acceptance Criteria ##
+
+Plugins must adhere to a number of conventions, mostly in the way the plugin files are laid out. This allows tools such as the `artifactory-user-plugins-devenv` and our internal testing system to properly recognize and work with plugins. These conventions are as follows:
```

Comment:
Give a more detailed why here.  "our internal testing system that is regularly run against all plugins in this repository" or somesuch.  Don't go too far in making a commitment, but making clear that we aren't just asking for tests but are actually using them...